internal/plugin/connectors/tcp/mssql/connection_details.go

```diff
@@ -8,50 +8,64 @@ import (
 // ConnectionDetails stores the connection info to the real backend database.
 // These values are pulled from the SingleUseConnector credentials config
 type ConnectionDetails struct {
-	Host     string
-	Port     uint
-	Username string
-	Password string
-	SSLMode  string
+	Host       string
+	Port       uint
+	Username   string
+	Password   string
+	SSLMode    string
+	SSLOptions map[string]string
+}
+
+var sslOptions = []string{
+	"sslrootcert",
+	"sslkey",
+	"sslcert",
 }
 
 const defaultMSSQLPort = uint(1433)
+const defaultSSLMode = "disable"
 
 // NewConnectionDetails is a constructor of ConnectionDetails structure from a
 // map of credentials.
 func NewConnectionDetails(credentials map[string][]byte) *ConnectionDetails {
 
 	connDetails := &ConnectionDetails{}
 
-	if host := credentials["host"]; host != nil {
+	if len(credentials["host"]) > 0 {
 		connDetails.Host = string(credentials["host"])
 	}
 
 	connDetails.Port = defaultMSSQLPort
-	if credentials["port"] != nil {
+	if len(credentials["port"]) > 0 {
 		port64, _ := strconv.ParseUint(string(credentials["port"]), 10, 64)
 		connDetails.Port = uint(port64)
 	}
 
-	if credentials["username"] != nil {
+	if len(credentials["username"]) > 0 {
 		connDetails.Username = string(credentials["username"])
 	}
 
-	if credentials["password"] != nil {
+	if len(credentials["password"]) > 0 {
 		connDetails.Password = string(credentials["password"])
 	}
 
 	// More supported cases to be added
 	sslMode := credentials["sslmode"]
 	switch {
-	case bytes.Equal(sslMode, []byte("disable")):
-		connDetails.SSLMode = string(credentials["sslmode"])
-	case bytes.Equal(sslMode, []byte("enable")):
-		fallthrough
+	case bytes.Equal(sslMode, []byte("require")):
+		connDetails.SSLMode = defaultSSLMode
+		break
 	case bytes.Equal(sslMode, []byte("verify-ca")):
 		fallthrough
+	case bytes.Equal(sslMode, []byte("disable")):
+		fallthrough
 	default:
-		connDetails.SSLMode = "disable"
+		connDetails.SSLMode = defaultSSLMode
+		return connDetails
+	}
```

Comment:
After some thought, the switch here seems unnecessary. I think our main goal right now is to force sslMode to disable always, and in the hypothetical (but not currently possible) case when sslMode is not disable we invoke parseSSLOptions. That can be achieved with:
```
sslMode := string(credentials["sslmode"])
if sslMode != "disable" {
	sslMode = "disable"
	connDetails.parseSSLOptions(credentials)
}

connDetails.SSLMode = sslMode
```

The switch seems to be anticipate something to come. I think the confusion/complexity that arises is more costly than the value we'll realise by preempting the code now. The YAGNI principle comes to mind here.

Reply:
also, can we have a debug message if we override what's written in the config with `disable`? that seems like something an end user might want to track

Reply:
The idea for the switch was to prevent values besides disable, moreso than preempting functionality. However this is a definitely better way to do this, so I'm happy to make the change.